rfcs/2021-03-02-6600-vector-remap-language-web.md

```diff
@@ -0,0 +1,169 @@
+# RFC 6600 - 2021-03-03 - Vector Remap Language on the web
+
+From the beginning of our work on [Vector Remap Language][VRL] (VRL), we've had discussions about
+novel ways to make VRL approachable and even downright fun for users. Thus far, we've done some
+pretty unique things, like providing a REPL for the language. But we've also had plans, though
+fairly unspecific ones, about making VRL available on the web in some form. This RFC seeks to lend
+some concreteness to those discussions.
+
+## Options
+
+In this RFC, I discuss three possibities. The first two emerged from our internal discussions, while
+the third was gathered from a survey of the Wasm ecosystem.
+
+1. An in-browser [web app][web_app]/playground for experimenting with and learning VRL
+2. A [web server][web_server] that can process VRL programs and events
+3. A [published Wasm module], largely for internal use
+
+I propose pursuing #1 and #3 in the near term, while leaving #2 as an option for the future.
+
+### Interactive web app
+
+An interactive VRL web app would be a browser-based app that provides an aesthetically pleasing and
+UX-friendly experience for learning and experimenting with VRL. In its most basic form, it would
+enable you to enter a VRL event and a VRL program into a text area and see what happens. But there
+are other capabilities that we should consider:
+
+* A step-by-step tutorial, like the one under way in PR [6184] that provides real exercises with
+  right and wrong answers
+* A built-in "library" of example events (JSON, Syslog, AWS formats, etc.)
```

Comment:
Yes, I see more "examples" that are selectable via drop down in the other playgrounds out there.  This is a lot simpler to implement and less friction for the user vs a step-by-step tutorial.  You select an example and loads up the text boxes with the input event and a sample script.  You can put inline comments in the example to explain what is being demonstrated.